test/object_store/conftest.py

```diff
@@ -24,11 +22,24 @@ def pytest_addoption(parser):
     s3_options.addoption('--s3-server-bucket')
 
 
-@dataclass
 class S3_Server:
-    address: str
-    port: int
-    bucket_name: str
+    def __init__(self, tempdir: str, address: str, port: int, bucket_name):
+        self.tempdir = tempdir
+        self.address = address
+        self.port = port
+        self.bucket_name = bucket_name
+        self.config_file = self._get_config_file()
+
+    def _get_config_file(self):
+        # if the test is started by test.py, minio_server.py should set this
+        # env variable for us, but if the test is started manually, there are
+        # chances that this env variable is not set, we would have to create it
+        # by ourselves, so the tests can consume it.
+        conffile = os.environ.get(MinioServer.ENV_CONFFILE)
+        if conffile is None:
+            conffile = os.path.join(self.tempdir, 'object-storage.yaml')
+            MinioServer.create_conf_file(self.address, self.port, conffile)
```

Comment:
But the whole point of generating config file from minio server was to free the test from this need. If test still may want to do it, then what's the gain?

Reply:
that's not true, the reason why i originally created this PR was to scratch my own itch -- to perform manual test against minio server. then per your suggestion, i expanded it so that the code creating the conffile can be reused and shared. the gain is

1. one can test the object store in a simpler way.
1. the code creating the conffile can be consolidated.